index.js

```
@@ -41,9 +41,24 @@ module.exports = function * (next) {
     return typeof item == 'number';
   });
 
+  //Adjust infinite end
+  if(end == Number.POSITIVE_INFINITY) {
+    if(this.length) {
+      end = this.length-1;
+    } else {
+      // This is ugly but we have to set something
+      end = 10000000000;
```

> The other alternative is to error out.

Yea, maybe it's better to throw a error